processing/src/main/java/io/druid/segment/Metadata.java

```diff
@@ -137,19 +164,25 @@ public boolean equals(Object o)
 
     Metadata metadata = (Metadata) o;
 
-    if (!container.equals(metadata.container)) {
+    if (container != null ? !container.equals(metadata.container) : metadata.container != null) {
```

Comment:
equals/hashCode are generated by intelliJ, but i think i can configure the generator to not do null check on container.